Multiple contours, hatches, and fhr added to ncl_name

christinaholtNOAA opened the pull request:
These changes add some new capabilities:

- An indefinite number of contours at various levels can be added to a map.
- Hatched areas can be added to a map.
- Forecast hour can be included as a format string in ncl_name with curly brackets. 

New tests:
- transforms are now tested to make sure that key word arguments are allowed for the functions listed.

Code is linted and tests pass.

cshartsough commented on adb_graphics/figures/maps.py:
```diff
@@ -118,14 +120,16 @@ class DataMap():
 
         field             datahandler data object for data field to shade
-        contour_field     dandhandler data object for data field to contour
+        contour_fields    list of dandhandler data objects fields to contour
```
"dandhandler" -> "datahandler"

christinaholtNOAA commented:
Oh. I suppose I also added all the hatched areas to plots where that was needed. And some of the contours.
[ADB_Graphics_PR46.pptx](https://github.com/NOAA-GSD/adb_graphics/files/5077005/ADB_Graphics_PR46.pptx)

cshartsough commented on adb_graphics/specs.py:
```diff
@@ -173,7 +173,7 @@ def lcl_colors(self) -> np.ndarray:
 
         ctable = ctables.colortables.get_colortable(self.vspec.get('cmap')) \
-                    (range(60, 180, 6)) # rainbow
+                    (range(50, 180, 7)) # rainbow
```
was this change something intentional or just a hiccup from our various color table tests?  just curious.

christinaholtNOAA replied:
I was attempting to bump it into darker blues just a tad to get the white contour at 1000 mb a tad more noticeable. Is that okay?

cshartsough replied:
Yeah, looks good, I just thought I might have done something odd in one of my PRs that you were trying to correct.  :-)

christinaholtNOAA commented:
Fixed a couple of things. 

- Added contours that were missing in the first round (1000 J/kg cape)
- Made the contour/hatch title labels more useful.
- Merged in Craig's changes and tested/linted.
